vault.go

```diff
@@ -25,7 +25,11 @@ func getCredentialSetFromVault(host string) (map[string]string, error) {
 	}
 
 	var out = map[string]string{}
-	for k, v := range secret.Data {
+	data, ok := secret.Data["data"].(map[string]interface{})
+	if !ok {
+		return nil, errors.Errorf("secret Data does not contain data")
+	}
+	for k, v := range data {
```

Comment:
This breaks K/V v1 compatibility.

Reply:
It is indeed incompatible, the vault doc https://www.vaultproject.io/docs/secrets/kv/kv-v2#acl-rules 
should a git-credential-vault v2 be release?

Reply:
As long as both K/V versions are supported by Vault they should be supported here too…

So basically if I see it correctly it should be this:
```
if secret.Data.data is map[string]interface{}:
	use secret.Data.data
elif secret.Data is map[string]interface{}:
	use secret.Data
else:
	error
```

At least I don't see any other difference which should be relevant for us… 🤔

Reply:
updated, please review, tks a lot.